lmdeploy/pytorch/models/deepseek_vl2.py

```diff
@@ -0,0 +1,560 @@
+                new_weights[new_key] = loaded_weight
+                continue
+
+            if 'qkv' in name and 'vision' not in name:
+                param = params_dict[name]
+                q, k, v = param.weight_spliter(loaded_weight)
+                load_weight(param, q, shard_id='q')
+                load_weight(param, k, shard_id='k')
+                load_weight(param, v, shard_id='v')
+            else:
+                param = params_dict[name]
+                load_weight(param, loaded_weight)
+
+        self.language.load_weights(new_weights.items())
+
+    def get_input_processor(self) -> BaseModelInputProcessor:
+        """get input processor."""
+        return self.input_processor
+
+
+class DeepSeekVLV2InputProcessor(BaseModelInputProcessor):
+    """deepseek-vl2 input processor."""
+
+    def __init__(self, config: PretrainedConfig, dtype) -> None:
+        self.config = config
+        self.dtype = dtype
+        vision_config = config.vision_config
+        self.patch_size = vision_config.patch_size
+
+    def preprocess_input(self,
+                         input_ids: List[int],
+                         input_multimodals: List[Dict[str, Any]] = None,
+                         **kwargs) -> PreprocessInputResult:
+        """prepare multimodal input."""
+        if input_multimodals is None or len(input_multimodals) == 0:
+            return input_ids, input_multimodals
+
+        input_imgs = []
+        for input_mm in input_multimodals:
+            pixel_values = input_mm['pixel_values'].to(self.dtype)
+            offset = input_mm['offset']
+            image_token_id = input_mm.get('image_token_id', 0)
+            num_pad = input_mm['image_tokens']
+            images_spatial_crop = input_mm.get('images_spatial_crop', None)
+            images_seq_mask = input_mm.get('images_seq_mask', None)
+            if isinstance(num_pad, torch.Tensor):
+                num_pad = num_pad.item()
+
+            # in multi-round conversations, lmdeploy won't add bos_id for non-first-round messages
+            # but upstream deepseek-vl2 processor always add bos_id for each message, need to remove bos_id
+            if len(input_ids) != images_seq_mask.shape[0]:
+                images_seq_mask = images_seq_mask[1:]
+
+            mm_data = MultiModalTensor(data=pixel_values,
+                                       start=offset,
+                                       end=offset + num_pad,
+                                       meta=dict(image_token_id=image_token_id,
+                                                 images_spatial_crop=images_spatial_crop,
+                                                 images_seq_mask=images_seq_mask))
+
+            input_imgs.append(mm_data)
+
+        result = PreprocessInputResult(
+            input_ids=input_ids,
+            input_multimodals=dict(image=input_imgs),
+        )
+
+        return result
+
+
+AutoConfig.register('vision', VisionEncoderConfig)
```

Comment:
we may not need to register again since they are registered in upstream repo
https://github.com/deepseek-ai/DeepSeek-VL2/blob/3c2cd21c9f16be75c1a59033e38e82a9e1e0d964/deepseek_vl2/models/modeling_deepseek_vl_v2.py#L694